pages/landing.tsx

```diff
@@ -0,0 +1,82 @@
+import styled from '@emotion/styled';
+import Header from '@src/components/common/Header';
+import { useRouter } from 'next/router';
+import React, { useState, useEffect } from 'react';
+import Image from 'next/image';
+import planetImage from '@assets/planet_main.svg';
+import IntroduceSentence from '@src/components/common/IntroduceSentence';
+
+function landing() {
+  const router = useRouter();
+  const messages: string[] = [
+    '나만의 우주를 만들어보세요.\n행성이 감정을 담아둘거예요',
+    '함께 느꼈던 감정을 행성에 기록할 친구들을\nYouniverse로 초대할 수 있어요.',
+  ];
+  const [message, setMessage] = useState(messages[0]);
+
+  useEffect(() => {
+    let timer = setInterval(() => {
+      switch (message) {
+        case messages[0]:
+          setMessage(messages[1]);
+          break;
+        case messages[1]:
+          router.push('/onboarding');
+          break;
+      }
+    }, 1500);
+    return () => clearTimeout(timer);
+  }, [message]);
+
+  return (
+    <>
+      <Header />
+      <Styled.Main>
+        <Styled.TextWrapper>
+          <IntroduceSentence />
+          <p>{message}</p>
```

Comment:
messageIndex만 받아서 표현하는 것이 더 유연하고 확장성이 좋은 것같아용~
그리고 1500 같은 매직 넘버는 상수로 뺴면 유지보수가 쉬울것같아여

```suggestion
  const [messageIndex, setMessageIndex] = useState(0);

  useEffect(() => {
    let timer = setInterval(() => {
      setMessageIndex(messageIndex + 1);
      if (messages.length === messageIndex) {
        router.push('/onboarding');
      }
    }, 1500);
    return () => clearTimeout(timer);
  }, []);

  return (
    <>
      <Header />
      <Styled.Main>
        <Styled.TextWrapper>
          <IntroduceSentence />
          <p>{messages[messageIndex]}</p>
```

Reply:
@euijinkk 이가 써준 코드 보면 계속 messageIndex가 0이 찍히고 온보딩으로 라우팅이 안돼 이유가 뭘까?
```tsx
useEffect(() => {
  let timer = setInterval(() => {
    setMessageIndex(messageIndex + 1);
    console.log(messageIndex);
    if (messages.length === messageIndex) {
      router.push('/onboarding');
    }
  }, 1500);
  return () => clearTimeout(timer);
}, []);
```
<img width="334" alt="image" src="https://user-images.githubusercontent.com/71601985/161941186-f60b60fe-dd06-4426-94f3-895afdb7646d.png">

Reply:
```tsx
useEffect(() => {
    let timer = setInterval(() => {
      if (messages.length - 1 === messageIndex) {
        router.push('/onboarding');
      } else {
        setMessageIndex(messageIndex + 1);
      }
    }, 1500);
    return () => clearTimeout(timer);
  }, [messageIndex]);
```

일단 이렇게 해결은 했다만 저게 왜 안됐었는지 모르겠다 ,,

Reply:
앗 ㅋㅋ length는 1부어 시작하고
index는 0부터 시작해서 그런듯.

-1 해줘서 해결 잘됐네용!~

Reply:
@dannaward 저런 방식으로 해결했다면 setInterval보다 setTimeout이 더 적절한 것 같아요! 
```typescript
  useEffect(() => {
    setTimeout(() => {
      console.log(messageIndex);
      if (messages.length - 1 === messageIndex) {
        router.push('/onboarding');
      } else {
        setMessageIndex(messageIndex + 1);
      }
    }, 1500);
  }, [messageIndex]);
```
이렇게요! 
setInterval되고, clear되는 과정이 불필요하게 반복되고 있는데 동작이 결국 setTimeout과 동일할 것 같습니다

Reply:
추가적으로, 의진이가 말해준 방법이 안된 이유는 훅의 클로저 문제인 것 같아요. state값이 useEffect의 스코프 안에 갇혀있어서 업데이트가 안되는 것 같습니다. 이 부분을 해결하기 위해 사용할 수 있는 커스텀 훅이 있는데 이거 제가 자세하게 정리해서 포스팅할 예정이니 일단은 위와 같은 방식으로 넘어가도 좋을 것 같아요! 넘 고생했슴다 : D